src/tests/test_bigint.cpp

```diff
@@ -544,11 +544,17 @@ BOTAN_REGISTER_TEST("math", "bn_rshift", BigInt_Rshift_Test);
 
 Test::Result test_const_time_left_shift() {
    Test::Result result("BigInt const time shift");
-   const size_t bits = Test::run_long_tests() ? 2 * 4096 : 2048;
+   const size_t bits = Test::run_long_tests() ? 4096 : 2048;
+
+   auto rng = Test::new_rng("const_time_left_shift");
+
+   result.start_timer();
 
    Botan::BigInt a = Botan::BigInt::with_capacity(bits / sizeof(Botan::word));
    for(size_t i = 0; i < bits; ++i) {
-      a.set_bit(i);
+      if(rng->next_byte() & 1) {
+         a.set_bit(i);
+      }
```

Comment:
Off topic: should we introduce something like `rng->next_bit() -> CT::Choice`? Not sure whether that would be useful in production code.

Reply:
Maybe? I can't think of cases currently where we predicate on a single random bit (vs say the even-y bit of a EC point). Somewhere in the PQ algos perhaps?

Reply:
Not really. They all just pull a seed and then sample from KDFs or XOFs. I didn't find something with an ad-hoc search either.